p8e-sdk/src/main/kotlin/io/p8e/ContractManager.kt

```diff
@@ -328,17 +328,45 @@ class ContractManager(
 //            }
 //    }
 
+
+    /**
+     * Returns an existing Contract object that can be configured and executed.
+     *
+     * @param contractClazz subclass of P8eContract that represents the original contract
+     * @param executionUuid the execution id of the original contract
+     * @param isFragment flag identifying if the original contract execution was requested (true) or initiated (false)
+     */
     fun <T: P8eContract> loadContract(
-        clazz: Class<T>,
-        executionUuid: UUID
+        contractClazz: Class<T>,
+        executionUuid: UUID,
+        isFragment: Boolean
     ): Contract<T> {
         return Contract(
             this,
             client,
-            dehydrateSpec(clazz),
+            dehydrateSpec(contractClazz),
             client.getContract(executionUuid),
-            clazz,
-            contractClassExecutor(clazz),
+            contractClazz,
+            contractClassExecutor(contractClazz),
+            isFragment
+        )
+    }
+
+    /**
+     * Returns a new Contract object that can be configured and executed. This Contract represents a contract
+     * based on a previously executed Contract that did not complete successfully. The envelope will contain
```

Comment:
Isn't this comment backwards? Isn't this for loading a previously executed contract (i.e. first step executed successfully, loading the contract for the second step execution)

Reply:
ahh yep, I'll make the update